src/js/src/services/RequestService.js

```diff
@@ -7,17 +7,33 @@ export const requestService = {
   fireImagesRequest
 }
 
-function fireSearchRequest (query, facets) {
+function fireSearchRequest (query, facets, options) {
+  let optionString = '&start=' + options.offset + '&grouping=' + options.grouping
   // Split url and move to config
-  const url = 'services/frontend/solr/search/results/' + `?query=${query + facets}`
+  const url = 'services/frontend/solr/search/results/' + `?query=${query + facets + optionString}`
   return axios.get(
     url, {
       transformResponse: [
         function(response) {
           let returnObj = JSON.parse(response)
-          for(let i = 0; i < returnObj.response.docs.length; i++) {
-            returnObj.response.docs[i].highlight = returnObj.highlighting[returnObj.response.docs[i].id]
+          if(options.grouping === false) {
+            for(let i = 0; i < returnObj.response.docs.length; i++) {
+              returnObj.response.docs[i].highlight = returnObj.highlighting[returnObj.response.docs[i].id]
+              }
           }
+          else {
+            returnObj.response = {}
+            returnObj.response.docs = []
+            returnObj.response.numFound = returnObj.grouped.url.doclist.numFound
+            returnObj.response.maxScore = returnObj.grouped.url.doclist.maxScore
+            returnObj.response.start = returnObj.grouped.url.doclist.start
+            returnObj.response.cardinality = returnObj.stats.stats_fields.url.cardinality
+            for(let i = 0; i < returnObj.grouped.url.doclist.docs.length; i++) {
+              returnObj.response.docs[i] = returnObj.grouped.url.doclist.docs[i]
+              returnObj.response.docs[i].highlight = returnObj.highlighting[returnObj.grouped.url.doclist.docs[i].id]
+            }
+          }
+          console.log(returnObj)
```

Comment:
Good point. I'll move it out into a seperate file - because we might get more stuff like this, seeing as we get different types of backend responses (ie images and what not)